packages/typescript-estree/src/convert.ts

```diff
@@ -539,27 +579,14 @@ export class Converter {
         result = this.createNode<TSESTree.JSXMemberExpression>(node, {
           type: AST_NODE_TYPES.JSXMemberExpression,
           object: this.convertJSXTagName(node.expression, parent),
-          property: this.convertJSXTagName(
-            node.name,
-            parent,
-          ) as TSESTree.JSXIdentifier,
+          property: this.convertJSXIdentifier(node.name),
         });
         break;
 
       case SyntaxKind.ThisKeyword:
-        result = this.createNode<TSESTree.JSXIdentifier>(node, {
-          type: AST_NODE_TYPES.JSXIdentifier,
-          name: 'this',
-        });
-        break;
```

Comment:
i'm unsure how we want to parse this, as this is invalid

i followed lead of babel on this 
```
<this:x />
```

https://astexplorer.net/#/gist/270cbd05d523f900d08730adce14815d/4ed17997e784261e930cafe2188ff705a53786a2